middleware/draftAppealStoreMiddleware.js

```diff
@@ -247,9 +247,11 @@ const restoreAllDraftsState = async(req, res, next) => {
             const shimmed = {};
             const drafts = result.body;
             const draftObj = {};
-
-            for (const draft of drafts) {
-              draftObj[draft.ccdCaseId] = draft;
+            if (Array.isArray(result.body) && result.body.length > 0) {
```

Comment:
You can directly check the drafts here instead of checking result.body
and use if(Array.isArray(drafts) && !draft.length)

Reply:
Modified